### PR TITLE
Use authenticated context

### LINKED ISSUE #58: Refactor Auth for routes
- Landing page should not have "get started" button when logged in 
- Maybe we can convert the Auth stuff to a context s.t. all components can access it

### PR DESCRIPTION
**Pull request by armiantos**
# Overview

Uses authenticated context to hide get started button when user is logged in. Snapshot files are automatically generated using jest. 

Enzyme is used for shallow rendering so that we can better "unit test" rather than rendering nested components

# Changes

* Hides get started button when user is authenticated
* Internal links now properly use `react-router-dom` links which doesn't require another http fetch to the web server. [Learn more about this here](https://material-ui.com/components/links/#third-party-routing-library)
* Use enzyme for UI testing

# Questions & Notes

# Issue tracking

Closes #58 

# Testing & Demo

* Check out the snapshot generated

| Logged out | Logged in |
| --- | --- |
| Screenshot_20210613_161533 | Screenshot_20210613_161554 |

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -48,12 +48,21 @@
...
         "@typescript-eslint/parser": "^4.25.0",
+        "@wojtekmaj/enzyme-adapter-react-17": "^0.6.2",
+        "enzyme": "^3.11.0",
+        "enzyme-to-json": "^3.6.2",
         "eslint": "^7.27.0",
         "eslint-plugin-prettier": "^3.4.0",
         "eslint-plugin-react": "^7.23.2",
         "prettier": "^2.3.0"
+    },
+    "jest": {
```
Comment (calebschoepp) [inline www/package.json:63]: Should this live in a `jest.config.js` file. This is the pattern we are following on the backend I believe.

File: www/src/routes/Landing.tsx [hunk 2]
```diff
@@ -55,11 +57,13 @@ const Intro: FC = () => {
...
                         </Grid>
-                        <Grid item>
-                            <Button variant='contained' color='primary' onClick={() => console.log('clicked Get Started button')}>
-                                {'Get Started'}
-                            </Button>
-                        </Grid>
+                        {!isAuthenticated && (
+                            <Grid item>
+                                <Button variant='contained' color='primary' onClick={() => console.log('clicked Get Started button')}>
+                                    {'Get Started'}
```
Comment (calebschoepp) [inline www/src/routes/Landing.tsx:63]: What is the point of the `Get Started` button? Why do we need to hide it when the user is logged in?

**Review by calebschoepp — COMMENTED** [review]
Good work. Left a couple comments. I don't feel like the final authority on `www` so I won't someone else to review this too.

File: www/package.json (hunk 1 above)
Comment (armiantos) [reply www/package.json:63]: I would like to but I think to make jest work for react, `create-react-app` has a preconfigured template and their guidance to override it is to put it in `package.json`. I tried exporting their default config but still no luck and am getting the following errors
image

File: www/src/routes/Landing.tsx (hunk 2 above)
Comment (armiantos) [reply www/src/routes/Landing.tsx:63]: I think its supposed to ask someone to log-in

File: www/package.json (hunk 1 above)
Comment (calebschoepp) [reply www/package.json:63]: Okay that is fine then.

File: www/src/routes/Landing.tsx (hunk 2 above)
Comment (cyrusdiego) [reply www/src/routes/Landing.tsx:63]: ^ yup, the `Get Started` button is just to bring them to the Login page

**Review by cyrusdiego — APPROVED** [review]
This LGTM! Good work on this :)

**Pull request merged**